x/httpproxy/path_handler.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpproxy
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type pathHandler struct {
+	client http.Client
+}
+
+var _ http.Handler = (*pathHandler)(nil)
+
+func (h *pathHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http.Request) {
+	pathReqString := strings.TrimPrefix(proxyReq.URL.Path, "/")
+
+	if pathReqString == "" {
+		http.Error(proxyResp, "Empty path", http.StatusBadRequest)
+		return
+	}
+
+	pathReqUrl, err := url.Parse(pathReqString)
+
+	if err != nil {
+		http.Error(proxyResp, "Invalid path", http.StatusBadRequest)
```

Comment:
I think this is clearer:
```suggestion
		http.Error(proxyResp, "Invalid target URL", http.StatusBadRequest)
```